cdap-common/src/main/java/io/cdap/cdap/internal/app/store/OperationRunDetail.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.store;
+
+import com.google.common.base.Objects;
+import com.google.gson.annotations.SerializedName;
+import io.cdap.cdap.proto.operationrun.OperationError;
+import io.cdap.cdap.proto.operationrun.OperationMeta;
+import io.cdap.cdap.proto.operationrun.OperationRun;
+import io.cdap.cdap.proto.operationrun.OperationRunStatus;
+import java.util.Arrays;
+import javax.annotation.Nullable;
+
+/**
+ * Store the meta information about operation runs in CDAP. This class contains all information the
+ * system needs about a run, which includes information that should not be exposed to users.
+ * {@link io.cdap.cdap.proto.operationrun.OperationRun} contains fields that are exposed to users,
+ * so everything else like the user request and principal goes here
+ *
+ * @param <T> The type of the operation request
+ */
+public class OperationRunDetail<T> extends OperationRun {
```

Comment:
Do you really want this inheritance? It makes Builder pretty cumbersome and it becomes easy to leak OperationRunDetail to user. It's better to make OperationRun a field or OperationRunDetail

Reply:
I agree the builder becomes a little cumbersome but once created should be fine. I chose inheritance as it follows the same pattern as ProgramRunDetail and the data model makes sense as we are only adding some new fields. 
The data leak is something that can occur if we are not careful. I will switch it to composition as there is no apparent major downside I can think of except the json field depth will increase while serialising.

Reply:
@albertshau Do you have any concern with the above ?

Reply:
no concern, +1 to making it a field instead of inheritance

Reply:
Updated to use composition